python/cudf/cudf/core/dataframe.py

```diff
@@ -398,18 +398,24 @@ def _init_from_list_like(self, data, index=None, columns=None):
             index = as_index(index)
 
         self._index = as_index(index)
-        data = list(itertools.zip_longest(*data))
 
-        if columns is not None and len(data) == 0:
-            data = [
-                cudf.core.column.column_empty(row_count=0, dtype=None)
-                for _ in columns
-            ]
+        # list-of-dicts case
+        if len(data) > 0 and np.all(isinstance(d, dict) for d in data):
```

Comment:
```suggestion
        if len(data) > 0 and isinstance(data[0], dict):
```
Maybe we can just make it more efficient by just checking the first object?

Reply:
I'm 👍  on that